pkg/plugins/resources/yaml/condition.go

```diff
@@ -16,18 +16,28 @@ func (y *Yaml) Condition(source string) (bool, error) {
 
 // ConditionFromSCM checks if a key exists in a yaml file
 func (y *Yaml) ConditionFromSCM(source string, scm scm.ScmHandler) (bool, error) {
-	if scm != nil {
-		y.spec.File = joinPathWithWorkingDirectoryPath(y.spec.File, scm.GetDirectory())
+	joignedFiles := make(map[string]string)
+	for filePath := range y.files {
+		joignedFilePath := filePath
+		if scm != nil {
+			joignedFilePath = joinPathWithWorkingDirectoryPath(joignedFilePath, scm.GetDirectory())
+			logrus.Debugf("Relative path detected: changing from %q to absolute path from SCM: %q", filePath, joignedFilePath)
+		}
+		joignedFiles[joignedFilePath] = y.files[filePath]
 	}
+	y.files = joignedFiles
 
 	return y.condition(source)
 }
 
 func (y *Yaml) condition(source string) (bool, error) {
+	var fileContent string
+	var filePath string
 
-	// Test at runtime if a file exist
-	if !y.contentRetriever.FileExists(y.spec.File) {
-		return false, fmt.Errorf("the yaml file %q does not exist", y.spec.File)
+	if len(y.files) > 1 {
```

Comment:
Is there any reason to not check for more than one condition? 
I can see some value here and I don't think it requires a lot of code change